anomalib/data/ucsd_ped.py

```diff
@@ -0,0 +1,269 @@
+"""UCSD Pedestrian dataset."""
+
+import glob
+import logging
+import os
+import tarfile
+from os import listdir, rmdir
+from os.path import join
+from pathlib import Path
+from shutil import move
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+from urllib.request import urlretrieve
+
+import albumentations as A
+import cv2
+import torch
+from pandas import DataFrame
+from torch import Tensor
+
+from anomalib.data.base import AnomalibDataModule
+from anomalib.data.base.video import VideoAnomalibDataset
+from anomalib.data.utils import (
+    DownloadProgressBar,
+    Split,
+    ValSplitMode,
+    hash_check,
+    read_image,
+)
+from anomalib.data.utils.video import ClipsIndexer
+from anomalib.pre_processing import PreProcessor
+
+logger = logging.getLogger(__name__)
+
+
+def make_ucsd_dataset(path: Path, split: Optional[Union[Split, str]] = None):
+    """Create UCSD Pedestrian dataset by parsing the file structure.
+
+    The files are expected to follow the structure:
+        path/to/dataset/category/split/video_id/image_filename.tif
+        path/to/dataset/category/split/video_id_gt/mask_filename.bmp
+
+    Args:
+        root (Path): Path to dataset
+        split (Optional[Union[Split, str]], optional): Dataset split (ie., either train or test). Defaults to None.
+
+    Example:
+        The following example shows how to get testing samples from UCSDped2 category:
+
+        >>> root = Path('./UCSDped')
+        >>> category = 'UCSDped2'
+        >>> path = root / category
+        >>> path
+        PosixPath('UCSDped/UCSDped2')
+
+        >>> samples = make_ucsd_dataset(path, split='test')
+        >>> samples.head()
+           root             folder image_path                    mask_path                         split
+        0  UCSDped/UCSDped2 Test   UCSDped/UCSDped2/Test/Test001 UCSDped/UCSDped2/Test/Test001_gt  test
+        1  UCSDped/UCSDped2 Test   UCSDped/UCSDped2/Test/Test002 UCSDped/UCSDped2/Test/Test002_gt  test
+        ...
+
+    Returns:
+        DataFrame: an output dataframe containing samples for the requested split (ie., train or test)
+    """
+    folders = [filename for filename in sorted(Path(path).glob("*/*")) if os.path.isdir(filename)]
+    folders = [folder for folder in folders if len(list(folder.glob("*.tif"))) > 0]
```

Comment:
Another suggestion by `refurb`
```suggestion
    folders = [folder for folder in folders if folder.glob("*.tif")]
```